### PR TITLE
python3.pkgs.kaitaistruct: add compression functionality

### PR DESCRIPTION
**Pull request by cole-h**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119992` at c945dbb8 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>mitmproxy (python38Packages.mitmproxy)</li>
    <li>python38Packages.kaitaistruct</li>
    <li>python39Packages.kaitaistruct</li>
    <li>python39Packages.mitmproxy</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/kaitaistruct/default.nix:35:3:
```
   |
35 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/kaitaistruct/default.nix:36:0:
```
   |
36 |     description = "Kaitai Struct: runtime library for Python";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/kaitaistruct/default.nix [hunk 1]
```diff
@@ -9,6 +23,15 @@ buildPythonPackage rec {
...
 
+  preBuild = lib.optionalString withCompress ''
+    ln -s ${kaitai_compress}/python/kaitai kaitai
+    sed '28ipackages = kaitai/compress' -i setup.cfg
+  '';
+
+  propagatedBuildInputs = lib.optionals withCompress [
+    lz4
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/kaitaistruct/default.nix:34]:
```suggestion
missing checkPhase and pythonImportsCheck.
```

File: pkgs/development/python-modules/kaitaistruct/default.nix [hunk 2]
```diff
@@ -9,6 +23,15 @@ buildPythonPackage rec {
...
   meta = with lib; {
     description = "Kaitai Struct: runtime library for Python";
     homepage = "https://github.com/kaitai-io/kaitai_struct_python_runtime";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/kaitaistruct/default.nix:37]: Maybe you want to add the maintainers team here, too.

File: pkgs/development/python-modules/kaitaistruct/default.nix [hunk 3]
```diff
@@ -1,5 +1,19 @@
-{ lib, buildPythonPackage, fetchPypi }:
+{ lib
+, buildPythonPackage
+, fetchPypi
+, fetchFromGitHub
+, withCompress ? false
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/kaitaistruct/default.nix:5]: Maybe we should enable that by default.

Comment (Mic92) [reply pkgs/development/python-modules/kaitaistruct/default.nix:5]: Agreed. I don't even think we need to make this an option.

File: pkgs/development/python-modules/kaitaistruct/default.nix (hunk 2 above)
Comment (grahamc) [reply pkgs/development/python-modules/kaitaistruct/default.nix:37]: Yeah, probably should since it is otherwise not maintained.

**Pull request merged**